integration_tests/src/main/python/date_time_test.py

```diff
@@ -137,55 +147,112 @@ def test_datediff(data_gen):
             'datediff(a, date(null))',
             'datediff(a, \'2016-03-02\')'))
 
-hms_fallback = ['ProjectExec'] if not is_supported_time_zone() else []
-
 @allow_non_gpu(*non_utc_tz_allow)
-def test_months_between():
+def test_months_between_runtime_fallback():
+    # We will do a CPU fallback during runtime for timezones with transitions during 
+    # years > 2200 as described in https://github.com/NVIDIA/spark-rapids/issues/6840
+    time_zone_gen = TimestampGen(end=last_supported_tz_time, tzinfo=ZoneInfo(os.environ.get('TZ', 'UTC')))
```

Comment:
Why are the tests for years > 2200 and for those before it using the same `end=last_supported_tz_time`?